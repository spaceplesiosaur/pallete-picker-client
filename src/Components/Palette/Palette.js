import React, { useState } from 'react';
import './Palette.scss';
import { FiLock, FiUnlock } from 'react-icons/fi';


const Palette = ({ name, color, frozen, freezeColor }) => {

  return (
    <section className='palette' style={{backgroundColor:`${color}`}} onClick={() => freezeColor(name)}>

      <FiLock />
      <p>HexCode: {color}</p>
    </section>
  )
};


export default Palette;
