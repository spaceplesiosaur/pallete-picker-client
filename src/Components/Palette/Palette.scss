.palette {
    height: 7em;
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    svg {
        font-size: 2em;
         margin: 2px;
         color: gray;
      }
}