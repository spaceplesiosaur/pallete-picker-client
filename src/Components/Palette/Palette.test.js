import React from 'react';
import { shallow } from 'enzyme';
import Palette from './Palette'

describe('Palette', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Palette
      freezeColor={jest.fn()}
      name={"The Eighties"}
      color={"#DFE9FD"}
      frozen={false}
      />)

  })

  it('should render with the correct properites', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
