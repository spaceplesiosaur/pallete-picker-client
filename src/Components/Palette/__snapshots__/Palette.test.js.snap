// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should render with the correct properites 1`] = `
<section
  className="palette"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#DFE9FD",
    }
  }
>
  <FiUnlock />
  <p>
    Hex-Code: 
    #DFE9FD
  </p>
</section>
`;
